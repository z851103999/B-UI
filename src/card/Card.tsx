import { computed, defineComponent, toRefs } from 'vue'
import { cardProps, CardProps } from './card-types'
import './card.less'

export default defineComponent({
  name: 'BCard',
  props: cardProps,
  setup(props: CardProps, { slots }) {
    const { align, src } = toRefs(props)
    const alignClass = computed(() => {
      return {
        'b-card__actions': true,
        [`b-card__action-align-${align.value}`]: align.value !== 'start',
      }
    })
    return () => (
      <div class={['card-container', 'b-card', `b-card__shadow-${props.shadow}`]}>
        {slots.default?.()}
        <div class="b-card__header">
          {slots.avatar?.() ? <div class="b-card__avatar">{slots.avatar?.()}</div> : ''}
          <div>
            <div class={'b-card__title'}>{slots.title?.()}</div>
            <div class={`b-card__subtitle`}>{slots.subtitle?.()}</div>
          </div>
        </div>
        {src.value !== '' ? <img src={src.value} alt="" class="b-card__meta" /> : ''}
        <div class="b-card__content">{slots.content?.()}</div>
        <div class={alignClass.value}>{slots.actions ? slots.actions?.() : ''}</div>
      </div>
    )
  },
})
