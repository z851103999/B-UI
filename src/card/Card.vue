<template>
  <div :class="[`b-card`, shadow ? `b-card__shadow` : null]">
    <slot name="image">
      <img
        class="b-card__image"
        :class="{ objectFit: fit, height: toSizeUnit(height) }"
        :src="src"
        :alt="alt"
        v-if="src"
      />
    </slot>
    <slot name="title">
      <div class="b-card__title" v-if="title">{{ title }}</div>
    </slot>
    <slot name="subtitle">
      <div class="b-card__subtitle" v-if="subtitle">{{ subtitle }}</div>
    </slot>
    <slot name="description">
      <div class="b-card__description" v-if="description">{{ description }}</div>
    </slot>
    <div class="b-card__footer" v-if="$slots.extra">
      <slot name="extra" />
    </div>
  </div>
</template>
<script lang="ts">
import { defineComponent } from 'vue'
import { props } from './props'
import { toSizeUnit } from '../utils/utils'

export default defineComponent({
  name: 'Card',
  props,
  setup() {
    return {
      toSizeUnit,
    }
  },
})
</script>

<style lang="less">
@import './card.less';
</style>
