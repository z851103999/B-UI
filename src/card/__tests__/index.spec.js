import example from '../example'
import Card from '..'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'

test('test card example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
})

test('test card plugin', () => {
  const app = createApp({}).use(Card)
  expect(app.component(Card.name)).toBeTruthy()
})
