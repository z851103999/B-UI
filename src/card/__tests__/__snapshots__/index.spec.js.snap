// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test card example 1`] = `
"<div class=\\"app-type\\"></div>
<div class=\\"b-card\\">
  <!--v-if-->
  <div class=\\"b-card__title\\">秋风</div>
  <div class=\\"b-card__subtitle\\">秋风subtitle</div>
  <div class=\\"b-card__description\\">秋风，把果实给了我们。风一吹，葡萄成熟了，晶莹剔透，犹如紫檀色的宝石上还闪着银白色的光呢！苹果结在枝头，红彤彤的，就像一盏盏红灯笼似的，点亮自己，照耀别人；黄黄的梨子宛如一个个金灿灿的葫芦；青绿的香瓜也逐渐在人们的口中传递着，棕褐色的籽、甜津津的水让人们赞不绝口；海南蕉也随着人们的喜爱上到了树上，等待人们采摘……秋风的魔力可真大啊！</div>
  <!--v-if-->
</div>
<div class=\\"app-type\\"></div>
<div class=\\"b-card b-card__shadow\\"><img class=\\"b-card__image objectFit\\" src=\\"https://p1-arco.byteimg.com/tos-cn-i-uwbnlip3yd/a20012a2d4d5b9db43dfc6a01fe508c0.png~tplv-uwbnlip3yd-webp.webp\\">
  <div class=\\"b-card__title\\">风景</div>
  <div class=\\"b-card__subtitle\\">公园路的风景</div>
  <div class=\\"b-card__description\\">秋风，把果实给了我们。风一吹，葡萄成熟了，晶莹剔透，犹如紫檀色的宝石上还闪着银白色的光呢！苹果结在枝头，红彤彤的，就像一盏盏红灯笼似的，点亮自己，照耀别人；黄黄的梨子宛如一个个金灿灿的葫芦；青绿的香瓜也逐渐在人们的口中传递着，棕褐色的籽、甜津津的水让人们赞不绝口</div>
  <div class=\\"b-card__footer\\"><button class=\\"b button primary\\" style=\\"margin-right: 10px;\\">添加按钮</button><button class=\\"b button warning\\">添加按钮</button></div>
</div>"
`;
